cmd/healthz/main.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/satellite/healthz/checks"
+	"github.com/gravitational/satellite/healthz/config"
+	"github.com/gravitational/satellite/healthz/handlers"
+	"github.com/gravitational/satellite/healthz/utils"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(trace.DebugReport(err))
+		fmt.Printf("ERROR: %v\n", err.Error())
+		os.Exit(255)
+	}
+}
+
+func run() error {
+	cfg := config.Config{}
+	config.ParseCLIFlags(&cfg)
+
+	trace.SetDebug(cfg.Debug)
+	if cfg.Debug {
+		log.SetLevel(log.DebugLevel)
+	}
+	log.SetFormatter(&log.TextFormatter{})
+	log.SetOutput(os.Stderr)
+
+	log.Debugf("[%q] starting using config: %#v", utils.SourceFileAndLine(), cfg)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		exitSignals := make(chan os.Signal, 1)
+		signal.Ignore()
```

Comment:
I just got tripped by a problem with ignoring `SIGCHLD` - ignoring these will cause child process handling broken. I think it's better to explicitly ignore the signals you want to ignore and leave the rest with system defaults.